lesson/narrative_and_checkpoints/narrative-and-checkpoints-template.md

```diff
@@ -1,51 +1,417 @@
+3. Checkpoint: _Run test command in terminal to verify_
 
-Hint: _Insert optional but recommended hint text here._
+    Hint: _Run ```npm test``` in terminal_
 
 <hr>
 
-## Exercise 2: _Insert exercise title here._
+## Exercise 3: _Unit Test With Jest (Part 1)_
 
 ### Narrative:
 
+Now that Jest is installed and configured; it is time to set up our first unit test. Let's start with the container that will hold our testing logic, the ```test()``` function. 
+
+The test() function takes three parameters:
+1. `"description of desired behavior"`
+2. () => {}
+3. timeout
+
+The first parameter acts as a way to label and identify the test in the testing output. This is where we describe the tested behavior. The second parameter is a function that will hold the actual test logic; we will revisit this parameter further in the next exercise. The final parameter is an optional timeout represented in milliseconds. This optional parameter defaults to 5 seconds if no value is provided and specifies how long a test should wait before aborting if something is preventing the test from finishing. 
+
+Once we put all of the parameters together, we get the following function:
+
+```test("description of desired behavior", () => {})```
+
 ### Instructions:
 
-1. Checkpoint: _Insert checkpoint text here._
+1. Checkpoint: _Create an empty test function in the language_spoken.test.js file using the syntax covered above._
 
-Hint: _Insert optional but recommended hint text here._
+    Hint: _```test()```_
 
-2. Checkpoint: _Insert checkpoint text here._
+2. Checkpoint: _Describe the test as ```”my first unit test”```_
 
-Hint: _Insert optional but recommended hint text here._
+    Hint: _The first parameter should be a string that reads, ```“my first unit test”```_
+3. Checkpoint: _Declare an empty function for the function that holds testing logic_
 
-<hr>
+    Hint: _The second parameter should be ```()=>{}```_
+4. Checkpoint: _Declare the function using the default timeout_
+
+    Hint: _The third parameter should be left blank._
+
+## Exercise 4: _Unit Test With Jest (Part 2)_
+
+### Narrative:
+
+With our test() function container setup, it is time to finish our first unit test. Jest offers an intuitive and easy-to-set-up testing syntax that provides a smooth transition between lamens terms and jest testing logic. Let's take, for example, an application that can tell you how many milligrams of potassium a list of food contains, we’ll call it “potassiumCounter”. As we begin testing this application, we will want to make sure that each of its individual functions works properly. We will start by testing the ```itemExtractor()``` function. We expect this function to receive an entered list in the form of a string and output it as an object. Below we can see this expected behavior translated into the jest testing syntax:
+
+```js
 
-## Exercise 3: _Insert exercise title here._
+//file: potassium_counter.test.js
+const potassiumCounter = require('./potassium_counter')
```

Comment:
Will learners know CommonJS syntax? If the learning environment uses a recent version of Node then we can stick with ES6. That way the syntax will be the same as the React material.

Reply:
+1